### ISSUE TITLE
Exception on destroying of a hopper

### ISSUE THREAD

**Issue by enaut** [Rusketh/ComparableHoppers2#23]
This form is for submitting issues with Comparable hoppers 2 for Spigot, please make sure to fill at all questions properlly as failing to do so will prevent us from helping replicate/fixing your issue.
This issue tracker is for Comparbale Hopper 2 Buyers only, if you have an issue with the plugin on a server but you are not the server owner please have the server owner or the plugins buyer fill out this form.

    Q: What is your name on spigot?
    A: enaut

    Q: What is your Spigot build?
    A: This is a final build for 1.12.2

3.1) Q: What is your product?
E.G: CH2 Basic, CH2 Paid, CH2 Spawners...
A: CH2 Paid 2.5.5,

3.2) Q: What is your Product version?
E.G CH 2.3.1 for MC 1.12.x
A: Enabling ComparableHoppers2 vBasic 2.4.2 custom build from #21

    Q: Do you get any errors in console?

```
[13:05:32 WARN]: java.lang.NullPointerException
[13:05:32 WARN]:         at rusketh.com.github.hoppers.helpers.SQLHelper.removeEntry(SQLHelper.java:84)
[13:05:32 WARN]:         at rusketh.com.github.hoppers.tiledata.HopperTile.OnBlockBreak(HopperTile.java:671)
[13:05:32 WARN]:         at rusketh.com.github.hoppers.listeners.BlockListener$1.run(BlockListener.java:71)
[13:05:32 WARN]:         at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:76)
[13:05:32 WARN]:         at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:361)
[13:05:32 WARN]:         at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:739)
[13:05:32 WARN]:         at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:406)
[13:05:32 WARN]:         at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679)
[13:05:32 WARN]:         at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577)
[13:05:32 WARN]:         at java.lang.Thread.run(Thread.java:748)
[13:05:32 WARN]: java.lang.NullPointerException
[13:05:32 WARN]:         at rusketh.com.github.hoppers.helpers.SQLHelper.removeEntry(SQLHelper.java:84)
[13:05:32 WARN]:         at rusketh.com.github.hoppers.tiledata.HopperTile.OnBlockBreak(HopperTile.java:673)
[13:05:32 WARN]:         at rusketh.com.github.hoppers.listeners.BlockListener$1.run(BlockListener.java:71)
[13:05:32 WARN]:         at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:76)
[13:05:32 WARN]:         at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:361)
[13:05:32 WARN]:         at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:739)
[13:05:32 WARN]:         at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:406)
[13:05:32 WARN]:         at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679)
[13:05:32 WARN]:         at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577)
[13:05:32 WARN]:         at java.lang.Thread.run(Thread.java:748)
[13:05:32 WARN]: java.lang.NullPointerException
[13:05:32 WARN]:         at rusketh.com.github.hoppers.helpers.SQLHelper.removeEntry(SQLHelper.java:84)
[13:05:32 WARN]:         at rusketh.com.github.hoppers.tiledata.HopperTile.OnBlockBreak(HopperTile.java:675)
[13:05:32 WARN]:         at rusketh.com.github.hoppers.listeners.BlockListener$1.run(BlockListener.java:71)
[13:05:32 WARN]:         at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:76)
[13:05:32 WARN]:         at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:361)
[13:05:32 WARN]:         at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:739)
[13:05:32 WARN]:         at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:406)
[13:05:32 WARN]:         at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679)
[13:05:32 WARN]:         at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577)
[13:05:32 WARN]:         at java.lang.Thread.run(Thread.java:748)
[13:05:32 WARN]: [ComparableHoppers2] Task #1598 for ComparableHoppers2 v2.5.5 generated an exception
java.lang.NullPointerException: null
        at rusketh.com.github.hoppers.tiledata.HopperTile.OnBlockBreak(HopperTile.java:677) ~[?:?]
        at rusketh.com.github.hoppers.listeners.BlockListener$1.run(BlockListener.java:71) ~[?:?]
        at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:76) ~[spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:361) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:739) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:406) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:679) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:577) [spigot.jar:git-Spigot-2cf50f0-2b93d83]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
```

5) Q: Have you read the full OP for the resource (https://www.spigotmc.org/resources/comparable-hoppers-2.45058/
   A: Yes

6) Q: Please describe the issue you are having.
   A: When destroying compareable hoppers this exception occurs. It does not affect gameplay. In case it matters the above exception was with a vanilla+ender hopper.

7) Q: Please describe how to replicate the issue, being as specific as possible.
   E.G Craft a comparable upgrade using the default recipie.
   A: see 6

**Comment by Rusketh** [comment]
Thank you, we are aware of this issue and it will be fixed hopefully in 2.5.6

**Comment by Rusketh** [comment]
Well, now that 2.5.6 is out for 1.13.1 we will begin attempting to port this version over to 1.12.2 for you, may take a little while but I want you know this issue inst being ignored.
